lib/lhm/entangler.rb

```diff
@@ -98,5 +101,21 @@ def revert
     def strip(sql)
       sql.strip.gsub(/\n */, "\n")
     end
+
+    def with_retry
+      begin
+        retries ||= 0
+        yield
+      rescue StandardError => e
+        if e.message =~ /timeout exceeded/ && retries < MAX_RETRIES
```

Comment:
Does this text match more than one error? How would about `e.errno == 1205` -- more exact but not as clear to read (it would need a comment :/)